### PR TITLE
Fixed #31597, #31603 -- Corrected admin clear all filters link behavior.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-31597
ticket-31603

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on selenium.

File: django/contrib/admin/views/main.py [hunk 1]
```diff
@@ -155,6 +158,7 @@ def get_filters(self, request):
                 # remove duplicate results.
                 if lookup_params_count > len(lookup_params):
                     use_distinct = use_distinct or lookup_needs_distinct(self.lookup_opts, field_path)
+                    has_active_filters = True
```
Comment (frnhr) [inline django/contrib/admin/views/main.py:161]: Not sure if my thinking if correct here, but it seems like we need to handle the `if callable(list_filter):` branch as well. I'm thinking of setting `has_active_filters = True` after the `else` and... and... blah blah!

django-contrib___Volumes_Projects_contrib_django-contrib__-__Volumes_Projects_contrib_django_django_contrib_admin_views_main_py__django_

Comment (felixxm) [reply django/contrib/admin/views/main.py]: Good catch :+1: I fixed this and added extra tests.

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Comment by felixxm** [comment]
> The tickets both complain that parameters are incorrectly removed. The change fixes that (testing manually) but I can't see that we're asserting that in the tests. Is it covered? Does it need to be?

We check in tests that only filters (`is_staff__exact`, `nr_of_members_partition`) are removed, see `href` in [R717-R729](https://github.com/django/django/pull/12934/files#diff-a7a67d6270c5d6e0ded64c97a77132ffR717-R729) and [R741-R749](https://github.com/django/django/pull/12934/files#diff-a7a67d6270c5d6e0ded64c97a77132ffR741-R749).

**Comment by carltongibson** [comment]
Perfect, yes. Clear all filters submits exactly that `data`. Thanks!

**Comment by felixxm** [comment]
buildbot, test on selenium.

**Pull request merged**